### DISCUSSION TITLE
Gradle / Maven Multi-Module support?

### DISCUSSION THREAD

**Discussion by matiwinnetou** [spring-projects/spring-modulith#692] [Q&A] [answered]
Does Spring Modulith support gradle / maven multi-module projects? The reason we want to go with this solution is because there are some parts that will be open sourced (modules with spring beans autoconfiguration and "starters") but project using those starters effectively needs to be private and not release in the open source way.

Is setup like this supported? I was thinking it consists of two tasks:
- split modulith project with existing modules into gradle / maven modules
- introduce spring starters as part of existing modules 
- introduce private project that depends on the released artefacts of starters (e.g. pushed to maven central).

Is such an architecture supported with Spring Modulith?

**Comment by matiwinnetou** [comment, marked as answer]
I will answer myself:

Actually I tried it and it fully works what is above.

**Reply by soulmz** [reply]
hi，How did you solve this problem?
